.github/workflows/reusable_linting_pure_python.yml

```diff
@@ -0,0 +1,110 @@
+name: reusable_linting_pure_python
+
+on:
+  workflow_dispatch:
+    inputs:
+      python_folder:
+        description: "Location of the python folder"
+        required: true
+        type: string
+      linting_folder:
+        description: "Location of the python folder to be linted"
+        required: true
+        type: string
+      windows:
+        description: "Run unittests on windows"
+        required: false
+        default: false
+        type: boolean
+      macos:
+        description: "Run unittests on macOS"
+        required: false
+        default: true
+        type: boolean
+  workflow_call:
+    inputs:
+      python_folder:
+        description: "Location of the python folder"
+        required: true
+        type: string
+      linting_folder:
+        description: "Location of the python folder to be linted"
+        required: true
+        type: string
+      windows:
+        description: "Run unittests on windows"
+        required: false
+        default: false
+        type: boolean
+      macos:
+        description: "Run unittests on macOS"
+        required: false
+        default: true
+        type: boolean
+
+jobs:
+
+  test_python_linting_linux:
+    name: python_check_linux-${{ matrix.python.interpreter }}
+    runs-on: "ubuntu-latest"
+    strategy:
+      matrix:
+        python: [
+            {py: '3.7', interpreter: "python3.7"},
+            {py: '3.8', interpreter: "python3.8"},
+            {py: '3.9', interpreter: "python3.9"},
+            {py: '3.10', interpreter: "python3.10"},
+            {py: '3.11', interpreter: "python3.11"},
+        ]
```

Comment:
Linting we should probably only do on one python version (I would prefer 3.9 because otherwise it might throw errors when we are not using features of later versions that we cannot use yet)

Reply:
Changed to 3.9